lib/queueClient.ts

```diff
@@ -165,4 +200,101 @@ export class QueueClient extends Client {
+   * LockDuration set on the Entity.
+   *
+   * @param {string | ServiceBusMessage} lockTokenOrMessage Lock token of the message or the message itself.
+   * @returns {Promise<Date>} Promise<Date> New lock token expiry date and time in UTC format.
+   */
+  async renewLock(lockTokenOrMessage: string | ServiceBusMessage): Promise<Date> {
+    return await this._context.managementClient!.renewLock(lockTokenOrMessage);
+  }
+
+  /**
+   * Schedules a message to appear on Service Bus at a later time.
+   *
+   * @param {SendableMessageInfo} message message that needs to be scheduled.
+   * @param scheduledEnqueueTimeUtc The UTC time at which the message should be available
+   * for processing.
+   * @returns {Promise<Long>} Promise<Long> The sequence number of the message that was
+   * scheduled.
+   */
+  async scheduleMessage(message: SendableMessageInfo, scheduledEnqueueTimeUtc: Date): Promise<Long> {
+    const scheduleMessages: ScheduleMessage[] = [
+      { message: message, scheduledEnqueueTimeUtc: scheduledEnqueueTimeUtc }
+    ];
+    const result = await this._context.managementClient!.scheduleMessages(scheduleMessages);
+    return result[0];
+  }
+
+  /**
+   * Schedules a message to appear on Service Bus at a later time.
+   *
+   * @param message - Message that needs to be scheduled.
+   * @param scheduledEnqueueTimeUtc - The UTC time at which the message should be available
+   * for processing.
+   * @returns Promise<number> - The sequence number of the message that was
+   * scheduled.
+   */
+  async scheduleMessages(messages: ScheduleMessage[]): Promise<Long[]> {
+    return await this._context.managementClient!.scheduleMessages(messages);
+  }
+
+  /**
+   * Cancels a message that was scheduled.
+   * @param sequenceNumber - The sequence number of the message to be cancelled.
+   * @returns Promise<void>
+   */
+  async cancelScheduledMessage(sequenceNumber: Long): Promise<void> {
+    return await this._context.managementClient!.cancelScheduledMessages([sequenceNumber]);
+  }
+
+  /**
+   * Cancels an array of messages that were scheduled.
+   * @param sequenceNumbers - An Array of sequence numbers of the message to be cancelled.
+   * @returns Promise<void>
+   */
+  async cancelScheduledMessages(sequenceNumbers: Long[]): Promise<void> {
+    return await this._context.managementClient!.cancelScheduledMessages(sequenceNumbers);
+  }
+
+  /**
+   * Receives a specific deferred message identified by `sequenceNumber` of the `Message`.
+   * @param sequenceNumber The sequence number of the message that will be received.
+   * @returns Promise<ServiceBusMessage | undefined>
+   * - Returns `Message` identified by sequence number.
+   * - Returns `undefined` if no such message is found.
+   * - Throws an error if the message has not been deferred.
+   */
+  async receiveDeferredMessage(sequenceNumber: Long): Promise<ServiceBusMessage | undefined> {
+    if (this.receiveMode !== ReceiveMode.peekLock) {
```

Comment:
It feels weird to be enforcing the receive mode here. These operations don't really use the receive mode in the way that the normal messaging operations do, so it feels like people should be allowed to use them regardless of configured mode.

Reply:
I see, what you are saying. Receivemode on the client has no bearing on receiving deferred messages from the $management endpoint. Disposing the message happens via updateDispositionStatus() on the $management endpoint. However, the updateDispositionStatus() happens within, defer(), complete(), abandon(), deadLetter() methods. The customer does not have to explicitly call updateDispositionStatus().

Now if the receiveMode is (let's say) ReceiveAndDelete, wont it be a little confusing for customers, if they are able to dispose off messages using message.defer|complete|abandon|deadletter()? 


I saw the same behavior in .net sdk. It [throws an error, if the receive mode is not peek lock](https://github.com/Azure/azure-service-bus-dotnet/blob/dev/src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs#L434). Hence, replicated the behavior over here for consistency.

Reply:
Ah okay. Hadn't dealt with the `updateDispositionStatus()` stuff before. If that's the usage then I think it makes sense to constrain that one to avoid weird errors

Reply:
Didn't see the comment about the .NET SDK before. I'm alright with keeping things this way in the name of consistency.